framework/include/times/Times.h

```diff
@@ -0,0 +1,76 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// Moose includes
+#include "GeneralReporter.h"
+
+/**
+ * Times objects are under the hood Reporters, but limited to a vector of Real
+ */
+class Times : public GeneralReporter
+{
+public:
+  static InputParameters validParams();
+  Times(const InputParameters & parameters);
+  virtual ~Times() = default;
+
+  /// Getter for the full times vector
+  const std::vector<Real> & getTimes() const;
+
+  /// Getter for a set of the full times
+  const std::set<Real> getUniqueTimes() const
+  {
+    return std::set<Real>(_times.begin(), _times.end());
+  };
+
+  /// Get the current time
+  Real getCurrentTime() const { return _fe_problem.time(); }
+
+  /// Getter for a single time at a known index
+  Real getTimeAtIndex(unsigned int index) const;
+
+  /// Find the previous time in the times vector for a given time
+  /// If current_time is also in the times vector within numerical precision, will return the previous value
+  Real getPreviousTime(const Real current_time) const;
+
+  /// Find the next time in the times vector for a given time
+  /// If current_time is also in the times vector within numerical precision, will return the next value
+  Real getNextTime(const Real current_time) const;
+
+protected:
+  /// In charge of computing / loading the times, unless all that could be done there is done
+  /// in the constructor
+  virtual void initialize() override = 0;
```

Comment:
If it's protected in the base, let's make this protected in the derived classes as well

Reply:
I think they are public in the base?

Reply:
This is `protected`. It says so right here. Derived classes of `Times` are overriding it in the public access space though

Reply:
fixed in derived classes